jupyter/notebook-controller/base/kustomization.yaml

```diff
@@ -15,7 +15,7 @@ commonLabels:
 images:
   - name: gcr.io/kubeflow-images-public/notebook-controller
     newName: gcr.io/kubeflow-images-public/notebook-controller
-    newTag: v20190911-e8193317-dirty-cd2831
+    digest: sha256:6490f737000bd1d2520ac4b8cbde2b09749cdb291b1967ddda95d05131db49db
```

Comment:
I think the 0911 should also be the latest.
Btw why use digest?
I think tag with the date should be more informative?

Reply:
shas are immutable whereas tags are not.
Using tags has a couple side effects

1) The tag names often end up including information we don't want

     * E.g; do we really want dirty in the tag?
     * If you rebuild the image using GCB from head you end up getting the following tag
       
       ```
       v20191007-v0.6.0-rc.0-134-g6ddf28cc-dirty-560d95 
       ```

     * This would likely confuse people because it says v0.6.0 but really its a v0.7.0 image
     * So one way to avoid that type of confusion is to just use the sha we can then map the sha to additional meta information as needed
    * e.g [http://gcr.io/kubeflow-images-public/notebook-controller@sha256:6490f737000bd1d2520ac4b8cbde2b09749cdb291b1967ddda95d05131db49db](http://gcr.io/kubeflow-images-public/notebook-controller@sha256:6490f737000bd1d2520ac4b8cbde2b09749cdb291b1967ddda95d05131db49db)  takes you to the GCR page for the image where you can see all the metadata infromation

1. We are trying to create automatic workflows to update the images and this will likely be easier to make work with the shas.

Reply:
got it, thanks for the explanation